model/lifecycle.go

```diff
@@ -1065,6 +999,54 @@ func createTasksForBuild(project *Project, buildVariant *BuildVariant, b *build.
 	return tasks, nil
 }
 
+func makeDeps(t BuildVariantTaskUnit, thisTask *task.Task, taskIds TaskIdTable) []task.Dependency {
+	dependencySet := make(map[task.Dependency]bool)
+	for _, dep := range t.DependsOn {
+		status := evergreen.TaskSucceeded
+		if dep.Status != "" {
+			status = dep.Status
+		}
+
+		// set unspecified fields to match thisTask
+		if dep.Name == "" {
```

Comment:
What's the reason for these defaults? Won't they just create a circular dependency if it gets hit?

Reply:
I was looking to preserve the [current behavior](https://github.com/evergreen-ci/evergreen/blob/3e4287f95e0566c11cbaa3ba138276eec4739b08/model/lifecycle.go#L942) where variant is set to the task's variant when no variant is provided. It's also what we currently do when [variant == "*" and the task isn't set](https://github.com/evergreen-ci/evergreen/blob/3e4287f95e0566c11cbaa3ba138276eec4739b08/model/lifecycle.go#L953) and when [task == "*" and variant isn't set](https://github.com/evergreen-ci/evergreen/blob/968f04c3b73a15d377e8c79ee3574cc6ba2dca14/model/lifecycle.go#L915).

I don't see how it would end up with a circular dependency, though: it won't depend on itself since we filter it out later.